test/evergreen/tcmalloc_install_or_build.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+#
+# Make a tcmalloc shared library avilable for preloading into the environment.
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 build-variant"
+    exit 1
+fi
+
+build_variant=$1
+
+# This is tcmalloc upstream revision 093ba93 source patched by the SERVER team.
+# https://github.com/mongodb-forks/tcmalloc/releases/tag/mongo-SERVER-85737
+PATCHED_SRC=mongo-SERVER-85737
+PATCHED_TGZ="${PATCHED_SRC}.tar.gz"
+PATCHED_TGZ_URL="https://github.com/mongodb-forks/tcmalloc/archive/refs/tags/${PATCHED_TGZ}"
+PATCHED_SRC_DIR="tcmalloc-${PATCHED_SRC}"
+
+# Sensitive variables retrieved through Evergreen so they won't be exposed
+# in source control.
+#
+# Export the AWS_* variables so that they are visible for the 'aws' cli.
+S3_URL=${s3_bucket_tcmalloc}
+export AWS_ACCESS_KEY_ID=${s3_access_key}
+export AWS_SECRET_ACCESS_KEY=${s3_secret_key}
+
+# Directory into which shared object will be installed.
+install_dir=$PWD
+tcmalloc_dir=TCMALLOC_LIB
+tcmalloc_so_dir=${install_dir}/${tcmalloc_dir}
+
+# Location in AWS S3 for prebuilt tcmalloc binaries.
+PREBUILT_TGZ="tcmalloc-${PATCHED_SRC}-${build_variant}.tgz"
+PREBUILT_URL="${s3_bucket_tcmalloc}/build/wt_prebuilt_tcmalloc/${PATCHED_SRC}/${PREBUILT_TGZ}"
+
+# Exit script unless command returns one of the specified error codes.
+# Command error code will be returned if not exited.
+# Usage: die_unless "cmd args..." ret rets...
+exit_unless() {
+        $1
+        ret=$?
+        for ret_ok in "${@:2}"; do
+                [ ${ret_ok} -eq ${ret} ] && return $ret
+        done
+        echo "$1 failed with code $ret"
+        exit $ret
+}
+
+# Without the --quiet option 'aws s3 cp' will print out an error message if the file is
+# NOT present. This is an expected error case that is handled later in this script, but 
+# emitting messages with "Error" in them, into the Evergreen log will cause confusion.
+exit_unless "aws s3 cp --quiet ${PREBUILT_URL} ${PREBUILT_TGZ}" 0 1
```

Comment:
This bit looks over-engineered and unsafe.

Evaluating a command from a string doesn't properly handle special characters and spaces that can potentially reside in the arguments.

Since `exit_unless` is only used once, it's a little bit early to generalise. I'd just test `$?` on the spot and act accordingly. Can use either `case` or `if / elif`.

Reply:
Agreed especially given the single use. This was used more extensively previously before restructuring the script.